src/ert/job_queue/job_queue_node.py

```diff
@@ -196,6 +219,62 @@ def run_timeout_callback(self) -> None:
         if self.callback_timeout:
             self.callback_timeout(*self.callback_arguments)
 
+    # this function only works on systems where multiprocessing.Process uses forking
+    def run_done_callback_forking(self) -> Tuple[LoadStatus, str]:
+        # status_msg has a maximum length of 1024 bytes.
+        # the size is immutable after creation due to being backed by a c array.
+        status_msg: "SynchronizedString" = mp.Array("c", b" " * 1024)  # type: ignore
+        callback_status: "Synchronized[c_int]" = mp.Value("i", 2)  # type: ignore
+        pcontext = ProcessWithException(
+            target=self.done_callback_wrapper,
+            kwargs={
+                "callback_arguments": self.callback_arguments,
+                "callback_status_shared": callback_status,
+                "status_msg_shared": status_msg,
+            },
+        )
+        pcontext.start()
+        try:
+            pcontext.wait_and_throw_if_exception()
+        except Exception as err:  # pylint: disable=broad-exception-caught
+            exception_with_stack = "".join(
+                traceback.format_exception(type(err), err, err.__traceback__)
+            )
+            error_message = (
+                "got exception while running forward_model_ok "
+                f"callback:\n{exception_with_stack}"
+            )
+            print(error_message)
+            logger.exception(err)
+        pcontext.join()
+
+        load_status = LoadStatus(callback_status.value)
+
+        # this step was added because the state_map update in
+        #      forward_model_ok does not propagate from the spawned process.
+        run_arg = self.callback_arguments[0]
+        run_arg.ensemble_storage.state_map[run_arg.iens] = (
+            RealizationState.HAS_DATA
+            if load_status == LoadStatus.LOAD_SUCCESSFUL
+            else RealizationState.LOAD_FAILURE
+        )
+
+        return load_status, status_msg.value.decode("utf-8")
+
+    def done_callback_wrapper(
+        self,
+        callback_arguments: CallbackArgs,
+        callback_status_shared: "Synchronized[c_int]",
+        status_msg_shared: "SynchronizedString",
+    ) -> None:
+        callback_status: Optional[LoadStatus]
+        status_msg: str
+        callback_status, status_msg = self.done_callback_function(*callback_arguments)
+
+        if callback_status is not None:
+            callback_status_shared.value = callback_status.value  # type: ignore
```

Comment:
This function is never called in tests. Is it still used?

Reply:
It should be called in tests, but it may not be picked up by coverage due to being run in a different process?

Reply:
done_callback_wrapper is the target of the ProcessWithException created in run_done_callback_forking